modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +41,618 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
+----
+
+==== Source
+
+The DataWeave script uses `payload.name` to select the
+value of the object that has the key `"name"`.
+Note that `payload.name` and `payload."name"` are both valid
+ways to perform the selection.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+The script outputs the String value of the input object, "Data Weave".
+
+[source,JSON,linenums]
+----
+"Data Weave"
+----
+
+=== Example
+
+On an Array, the selector applies to every element.
+
+
+==== Input
+
+The input payload is an array that contains two objects with the same
+key, "name".
+
+[source,JSON,linenums]
+----
+[
+  {
+    "name": "Arg"
+  },
+  {
+    "name": "Japan"
+  }
+]
+----
+
+==== Source
+
+The DataWeave script uses `payload.name` to select the
+value of any objects in input payload that have the key, "name".
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+The script outputs an array that contains the values of the input objects
+with the key "name": "Arg" and "Japan".
+
+[source,json, linenums]
+----
+["Arg", "Japan"]
+----
+
+[[namespace_example1]]
+== Namespace Selector
+
+The DataWeave namespace selector supports the use of a name that matches a
+namespace or a local part. To match a namespace, the name must match the
+namespace and the local part. If the name only matches the local part, the
+selector will match all namespaces with that local part, regardless of the
+namespace.
+
+Note that XML namespaces and their prefixes are defined in the `xmlns` attribute
+of an XML element. For example, the element
+`<h:table xmlns:h="http://www.w3.org/TR/html4/"/>` defines a namespace that
+assigns the prefix `h` to the namespace  `http://www.w3.org/TR/html4/`.
+All elements in that namespace contain the `h` prefix. For example, in the
+element `<h:table/>`, `h` is the prefix for the namespace, and `table` is the
+local part.
+
+=== Example
+
+This example selects values from XML elements using the namespace and local part.
+
+==== Input
+
+The input payload contains XML elements, `h:table` and `f:table`, that have
+different namespaces but the same local names (`table`).
+
+[source,xml, linenums]
+----
+<root>
+
+  <h:table xmlns:h="http://www.w3.org/TR/html4/">
+    <h:tr>
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </h:table>
+
+  <f:table xmlns:f="https://www.w3schools.com/furniture">
+    <f:tr>
+      <f:name>African Coffee Table</f:name>
+      <f:width>80</f:width>
+      <f:length>120</f:length>
+    </f:tr>
+  </f:table>
+
+</root>
+----
+
+[[namespace_variable]]
+==== Source
+
+The DataWeave script selects XML content from specified namespaces. The
+script's header creates namespace (`ns`) variables `html` and `furniture`
+to store namespaces from the input payload. To select the children of `h:table`
+into element `a` and the children of `f:table` elements into element `b`, the
```

Comment:
Do we need to use "select into" as a verb? Is this "move into"? Or "select and move into"?

Reply:
we select the value of `payload.root.html#table` into the an object with the key `a`, for example.